### PR TITLE
Persist sort order returned by normalizr

### PR DESCRIPTION
**Pull request by coffenbacher**
When data is returned from a REST API in an array
It often has a useful sort order
That should be available to client code.

This data is returned from the normalizr.normalize
Function as `result`, but currently only `entities`
Are persisted.

Addresses https://github.com/TheGnarCo/redux-entity-config/issues/17

### CODE REVIEW

**Comment by coffenbacher** [comment]
@mikestone14 @pachun can you take a look at this one? I can't request reviewers on this repo

**Comment by pachun** [comment]
This looks alright to me. @mikestone14 will have more context. The explanation by Dan Abramov which you liked to from the issue you created was really helpful. I'd like to keep that in the commit message. Otherwise, I think this is good to go pending Mike's approval.

**Review by pachun — APPROVED** [review]
This looks alright to me. @mikestone14 will have more context. The explanation by Dan Abramov which you liked to from the issue you created was really helpful. I'd like to keep that in the commit message. Otherwise, I think this is good to go pending Mike's approval.

**Review by mikestone14 — CHANGES_REQUESTED** [review]
Thanks for contributing @coffenbacher! A couple notes:

This seems to cover the case of the `loadAll` action in which we would want to replace `sortedIds` with `payload.data.sortedIds` in the reducer. I think this is missing the cases for `create`, `load`, and `destroy`.

## Scenarios:

**`create`**: The `resultIds` (containing just the newly created entity id) should be added to a new copy of the existing `resultIds` and replace `resultIds`.

**`load`**: If the loaded entity's id exists in `resultIds` we should do nothing with that key. If the loaded entity's id does not exist in `resultIds` we should add it to the end of the `resultIds` array.

**`destroy`**: We should remove the destroyed entity's id from from the `resultIds` array.

File: package.json
Comment (mikestone14) [inline package.json]: This shouldn't be updated as part of the PR but can be done as part of a new tagged commit for version update along with an update to the changelog.

Comment (coffenbacher) [reply package.json]: Okay cool, I guess I leave that to you then?

**Comment by coffenbacher** [comment]
@mikestone14 @pachun I added a new version that covers and tests the Create/Load/Destroy scenarios

**Comment by BlakeWilliams** [comment]
I don't think it'd be unreasonable to expect the client to sort the records it retrieves.

For example, storing each returned result by `id` in an object (for access/performance reasons when working with individual results) you'd want to do something like: `Object.values(records)` which I don't believe guarantees order.

Not saying we should or shouldn't, just throwing the thought of "maybe the client should be responsible for order" out there.

File: src/config.js
Comment (mikestone14) [inline src/config.js]: Should this be `payload.data.sortedIds`? This also assumes that the destroyed entity is returned by the server which is not always (maybe even rarely) the case. There is [another issue](https://github.com/TheGnarCo/redux-entity-config/issues/12) for handling deleting entities though so not in scope for this change.

File: src/config.js
Comment (mikestone14) [inline src/config.js:54]: Should this be `payload.data.entities[entityName]`?

**Review by mikestone14 — COMMENTED** [review]
Looking good @coffenbacher just a couple more comments. I'm also curious why the existing tests are passing if we're changing the shape of the payload in the reducer. I'd be happy to pair on this later if that would be helpful.

**Comment by darkmoves** [comment]
Looking into your question about existing tests, @mikestone14, the payload contents are only being tested in `config.tests.js` via method calls that don't hit the reducer. I wouldn't expect them to fail with this change.

**Comment by mikestone14** [comment]
@darkmoves you're right but I think that means we're missing test coverage.

File: src/config.js
Comment (darkmoves) [reply src/config.js:54]: Based on [this documentation](https://github.com/paularmstrong/normalizr/blob/master/docs/api.md#output) of normalizr, I think you're right. The code on L41 would need to be changed too. 

But I would expect tests to fail if this code is wrong. Will check out the tests now.

Comment (darkmoves) [reply src/config.js:54]: I checked out the test for this code, and it seems to reflect reality. I changed L54 to `...payload.data.entities[entityName]` and the test failed. Given those two things, I believe the code is correct as written. I'm not sure exactly where the `entities` key is removed from `payload.data`, possibly [here](https://github.com/TheGnarCo/redux-entity-config/pull/18/files#diff-fa32e59756e26177366b9f7645dfd757R123).

File: src/config.js
Comment (darkmoves) [reply src/config.js]: Yup, I think it should.

Comment (darkmoves) [reply src/config.js]: For posterity: no change to the `sortedIds` array for a deletion, since there's ambiguity as to whether an object will be returned with the API response.

File: src/config.js
Comment (darkmoves) [reply src/config.js:54]: Turns out, it's happening [here](https://github.com/TheGnarCo/redux-entity-config/pull/18/files#diff-fa32e59756e26177366b9f7645dfd757R125) when entities is deconstructed.

**Comment by darkmoves** [comment]
@mikestone14 commit pushed with the changes that we discussed. This should be good to go.

**Comment by darkmoves** [comment]
This PR's been rebased to include the DESTROY fix and the `sortedIds` array has been added for that action. Ready for another round.

**Review by mikestone14 — APPROVED** [review]

**Pull request merged**
